### ISSUE TITLE
Cross-thread operation not valid: Control '' accessed from a thread other than the thread it was created on.

### ISSUE THREAD

**Issue by vrgamestudio** [BluestormDNA/ProjectPSX#25]
Hello,

Your project looked interesting so I decided to try it out. These were the steps I took:

Using VS2019, I downloaded a zip file of the code base and loaded it up.

I altered all your csproj files to have

`<TargetFramework>net5.0</TargetFramework>`

or 

`<TargetFramework>net5.0-windows</TargetFramework>`

for the winforms project.

I then had to alter the file 

`ProjectPSX\Devices\Expansion\Exp2.cs`

to declare the namespace in the older fashion:

```
namespace ProjectPSX.Devices.Expansion {
    public class Exp2 {
```

I provided a copy of "SCPH1001.BIN" and ran the Winforms project, selecting a Ridge Racer .cue file. This was my result in the console window:

```
[MemCard] No Card found. Will try to generate a new one on save.
[CD Track Builder] Generating CD Tracks from: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA).cue
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 01).bin Size: 3683232 Number: 1 LbaStart: 150 LbaEnd: 1566 isAudio False
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 02).bin Size: 56419776 Number: 2 LbaStart: 1866 LbaEnd: 25554 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 03).bin Size: 48500592 Number: 3 LbaStart: 25854 LbaEnd: 46175 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 04).bin Size: 56567952 Number: 4 LbaStart: 46475 LbaEnd: 70226 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 05).bin Size: 57640464 Number: 5 LbaStart: 70526 LbaEnd: 94733 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 06).bin Size: 54481728 Number: 6 LbaStart: 95033 LbaEnd: 117897 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 07).bin Size: 53670288 Number: 7 LbaStart: 118197 LbaEnd: 140716 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 08).bin Size: 21523152 Number: 8 LbaStart: 141016 LbaEnd: 149867 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 09).bin Size: 13629840 Number: 9 LbaStart: 150167 LbaEnd: 155662 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 10).bin Size: 2065056 Number: 10 LbaStart: 155962 LbaEnd: 156540 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 11).bin Size: 21878304 Number: 11 LbaStart: 156840 LbaEnd: 165842 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 12).bin Size: 17449488 Number: 12 LbaStart: 166142 LbaEnd: 173261 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 13).bin Size: 26610528 Number: 13 LbaStart: 173561 LbaEnd: 184575 isAudio True
File: D:\Emulation\Roms\Playstation\Ridge Racer (USA)\Ridge Racer (USA) (Track 14).bin Size: 33059712 Number: 14 LbaStart: 184875 LbaEnd: 198631 isAudio True
Track 0 size: 3683232 lbaStart: 150 lbaEnd: 1566
Track 1 size: 56419776 lbaStart: 1866 lbaEnd: 25554
Track 2 size: 48500592 lbaStart: 25854 lbaEnd: 46175
Track 3 size: 56567952 lbaStart: 46475 lbaEnd: 70226
Track 4 size: 57640464 lbaStart: 70526 lbaEnd: 94733
Track 5 size: 54481728 lbaStart: 95033 lbaEnd: 117897
Track 6 size: 53670288 lbaStart: 118197 lbaEnd: 140716
Track 7 size: 21523152 lbaStart: 141016 lbaEnd: 149867
Track 8 size: 13629840 lbaStart: 150167 lbaEnd: 155662
Track 9 size: 2065056 lbaStart: 155962 lbaEnd: 156540
Track 10 size: 21878304 lbaStart: 156840 lbaEnd: 165842
Track 11 size: 17449488 lbaStart: 166142 lbaEnd: 173261
Track 12 size: 26610528 lbaStart: 173561 lbaEnd: 184575
Track 13 size: 33059712 lbaStart: 184875 lbaEnd: 198631
[BUS] BIOS File found. Contents Loaded.
[EXP2] PSX: POST [f]
[EXP2] PSX: POST [e]
[EXP2] PSX: POST [1]
[EXP2] PSX: POST [2]
[EXP2] PSX: POST [3]
[EXP2] PSX: POST [4]
[EXP2] PSX: POST [1]
[EXP2] PSX: POST [3]
[EXP2] PSX: POST [4]
[EXP2] PSX: POST [5]
[EXP2] PSX: POST [6]
[EXP2] PSX: POST [2]
[EXP2] PSX: POST [5]
[EXP2] PSX: POST [6]
[EXP2] PSX: POST [7]
System.InvalidOperationException: Cross-thread operation not valid: Control '' accessed from a thread other than the thread it was created on.
   at System.Windows.Forms.Control.get_Handle()
   at ProjectPSX.Devices.GPU.tick(Int32 cycles) in D:\Projects\Tests\ProjectPSX-master\ProjectPSX\Devices\GPU\GPU.cs:line 184
   at ProjectPSX.ProjectPSX.RunFrame() in D:\Projects\Tests\ProjectPSX-master\ProjectPSX\Core\ProjectPSX.cs:line 58
   at ProjectPSX.Window.EXECUTE() in D:\Projects\Tests\ProjectPSX-master\ProjectPSX.WinForms\UI\Window.cs:line 320
```

Any ideas how I could get this running?

EDIT: It runs if I Start Without Debugging. But it runs too slow in Debug Build and too fast in Release build.

**Comment by BluestormDNA** [comment]
Any specific reason why you are not using VS2022 and .NET7?

I'm aware that it crashes on debug but i never fixed it as i always run on release.
ProjectPSX is a personal project to learn about emulation. It is intended to run too fast. It's quite handy to test things faster.
If you are willing to do your own "game loop" you can just modify `EXECUTE()`

**Comment by BluestormDNA** [comment]
If you want to debug the crash is fixed on: a797a254314f3931aa2d50abc4d4eb694de51deb

**Comment by vrgamestudio** [comment]
Thank you so much, I'll give a797a25 a try!

I'm not using VS2022 and .NET 7 as I don't normally have need of them and so don't have those installed. There was nothing about your project that required the up to date version, other than that you said it did, so all good.
